src/modules/authorizer/RoleAuthorizer.sol

```diff
@@ -91,36 +71,32 @@ contract RoleAuthorizer is
         internal
         onlyInitializing
     {
-        // Note about DEFAULT_ADMIN_ROLE: The DEFAULT_ADMIN_ROLE has admin privileges on all roles in the contract. It starts out empty, but we set the proposal owners as "admins of the admin role",
+        // Note about DEFAULT_ADMIN_ROLE: The DEFAULT_ADMIN_ROLE has admin privileges on all roles in the contract. It starts out empty, but we set the orchestrator owners as "admins of the admin role",
         // so they can whitelist an address which then will have full write access to the roles in the system. This is mainly intended for safety/recovery situations,
         // Modules can opt out of this on a per-role basis by setting the admin role to "BURN_ADMIN_ROLE".
 
-        // Store RoleIDs for the Proposal roles:
-        PROPOSAL_OWNER_ROLE =
-            generateRoleId(address(proposal()), uint8(CoreRoles.OWNER));
-        PROPOSAL_MANAGER_ROLE =
-            generateRoleId(address(proposal()), uint8(CoreRoles.MANAGER));
+        //We preliminarily grant admin role to the caller
+        _grantRole(ORCHESTRATOR_OWNER_ROLE, _msgSender());
 
         // Set up OWNER role structure:
 
         // -> set OWNER as admin of itself
-        _setRoleAdmin(PROPOSAL_OWNER_ROLE, PROPOSAL_OWNER_ROLE);
+        _setRoleAdmin(ORCHESTRATOR_OWNER_ROLE, ORCHESTRATOR_OWNER_ROLE);
         // -> set OWNER as admin of DEFAULT_ADMIN_ROLE
-        _setRoleAdmin(DEFAULT_ADMIN_ROLE, PROPOSAL_OWNER_ROLE);
-
-        // grant OWNER role to user from configData.
-        // Note: If the initial owner is 0x0, it defaults to msgSender()
-        if (initialOwner == address(0)) {
-            _grantRole(PROPOSAL_OWNER_ROLE, _msgSender());
-        } else {
-            _grantRole(PROPOSAL_OWNER_ROLE, initialOwner);
-        }
+        _setRoleAdmin(DEFAULT_ADMIN_ROLE, ORCHESTRATOR_OWNER_ROLE);
 
         // Set up MANAGER role structure:
         // -> set OWNER as admin of DEFAULT_ADMIN_ROLE
-        _setRoleAdmin(PROPOSAL_MANAGER_ROLE, PROPOSAL_OWNER_ROLE);
+        _setRoleAdmin(ORCHESTRATOR_MANAGER_ROLE, ORCHESTRATOR_OWNER_ROLE);
         // grant MANAGER Role to specified address
-        _grantRole(PROPOSAL_MANAGER_ROLE, initialManager);
+        _grantRole(ORCHESTRATOR_MANAGER_ROLE, initialManager);
+
+        // If there is an initial owner specified, we set it as owner and remove the deployer
+        // Note: If the initial owner is 0x0, it stays as msgSender()
+        if (initialOwner != address(0)) {
+            _grantRole(ORCHESTRATOR_OWNER_ROLE, initialOwner);
+            renounceRole(ORCHESTRATOR_OWNER_ROLE, _msgSender());
```

Comment:
If initialOwner == _msgSender() this will renounce the role entirely so there should also be a check with the initialOwner != address(0) that initialOwner != _msgSender() otherwise this entire if should be skipped too.